src/Functional/Group.php

```diff
@@ -32,6 +32,11 @@ function group($collection, callable $callback)
 
         InvalidArgumentException::assertValidArrayKey($groupKey, __FUNCTION__);
 
+        // Avoid implicit precision-loss from doubles (which cannot be keys)
+        if (is_numeric($groupKey)) {
```

Comment:
Can you add a testcase for this so that we have the behavior explicitly pinned down?